ktor-server/ktor-server-core/jvm/src/io/ktor/sessions/SessionTransportTransformerEncrypt.kt

```diff
@@ -29,14 +29,18 @@ import javax.crypto.spec.*
 class SessionTransportTransformerEncrypt(
     val encryptionKeySpec: SecretKeySpec,
     val signKeySpec: SecretKeySpec,
-    val ivGenerator: (size: Int) -> ByteArray = { size -> SecureRandom().generateSeed(size) },
+    ivGenerator: ((size: Int) -> ByteArray)? = null,
     val encryptAlgorithm: String = encryptionKeySpec.algorithm,
     val signAlgorithm: String = signKeySpec.algorithm
 ) : SessionTransportTransformer {
     companion object {
         private val log = LoggerFactory.getLogger(SessionTransportTransformerEncrypt::class.qualifiedName)
     }
 
+    private val random by lazy { SecureRandom.getInstance("SHA1PRNG").apply { generateSeed(16) } }
```

Comment:
SHA-1 is considered cryptographically broken by NIST as of 2011. Does that same concern apply here?

Reply:
I'd just recommend using `new SecureRandom()`. It will be secure enough.